File: java/school/Student.java
package school;

import java.util.List;

public class Student extends Person {

  private List<Course> courses;

  public Student(String name, String email, List<Course> courses) {
    super(name, email);
    this.courses = courses;
  }

  public List<Course> getCourses() {
    return courses;
  }
}

File: cpp/Student.cpp
// Copyright (c) Younes Boubekeur, 2020

#include "Student.h"
#include "Course.h"

Student::Student(std::string name, std::string email, std::vector<Course> courses):
    Person(name, email) {
  m_courses = courses;
}

std::vector<Course> Student::GetCourses() {
  return m_courses;
}
